lib/omnibus/fetchers/net_fetcher.rb

```diff
@@ -130,6 +130,25 @@ def checksum
       source[digest_type]
     end
 
+    #
+    # The md5 equivalent checksum of the fetched assets.
+    #
+    # If the user provides an md5 in the software definition, then that is
+    # used. Otherwise, the downloaded file is verified using the user
+    # specified digest and a fresh md5 checksum is calculated.
+    #
+    # Remember to call this function with s3 caching disabled if you are
+    # attempting to generate this md5 to look up this object in s3.
+    #
+    # @return [String]
+    #
+    def md5
```

Comment:
The object will still be named by its checksum (sha1).  This is just to populate the md5 field when uploading to s3.

Now that I think about it though - you're right.  It might be better to move this function into s3_cache and just have it calculate the md5.  digest_type was not being exposed and I was hoping to initially mirror the working of "checksum" with "actually get the md5" but that's more confusing that having the s3_cache module simply recalculate the md5 unconditionally when uploading to s3.